back/app/services/profanity_service.rb

```diff
@@ -8,13 +8,11 @@ def search_blocked_words text
       locale.split('-').first
     end.uniq.flat_map do |lang|
       blocked_words = Rails.cache.fetch("#{lang}/blocked_words_list", expires_in: 1.hour) do
-        fetch_blocked_words(lang).map{|w| normalize_text w}
+        Set.new(fetch_blocked_words(lang).map{|w| normalize_text w})
       end
       words = without_special_chars(normalize_text(text)).split ' '
-      blocked_words.select do |blocked_word|
-        words.include? blocked_word
-      end.map do |blocked_word|
-        { 
+      (blocked_words & words).map do |blocked_word|
```

Comment:
we can revert https://github.com/CitizenLabDotCo/citizenlab/pull/479/commits/a57785e8710b5042360d1676ee9a9a4ba8cd9bb6 after the cache expires (1 hour)

Reply:
alternatively, we could revert it now and change the cache key to `"#{lang}/blocked_words_set"` or something like that